### PR TITLE
pending cancel flow

### PR DESCRIPTION
**Pull request by jjidt**
- pending-active users now get same cancellation flow as active users
- also updates subscription deletion hook to update data on org, and await the data handling functions to ensure success

### CODE REVIEW

File: api/src/coordinators/billing.js [hunk 1]
```diff
@@ -230,9 +230,12 @@ const cancelAccount = async function(orgId) {
...
   }
 
-  if (currentSubscription.status === STRIPE_STATUS.TRIALING) {
-    // still in trial with or without billing info, cancel immediately
+  const orgSubscriptionStatus = getOrgSubscriptionStatusFromStripeCustomer(customer)
+
+  if (orgSubscriptionStatus === ORG_SUBSCRIPTION_STATUS.TRIALING) {
+    // still in trial, but not pending active, cancel immediately
     await stripeIntegrator.cancelSubscription(currentSubscription.id)
+    // TODO immediately trigger deletion of org and data here
```
Comment (dheniges) [inline api/src/coordinators/billing.js]: Is there a ticket for this?

**Review by dheniges — COMMENTED** [review]
Code looks good, but I didn't see a ticket for the todo

**Comment by jjidt** [comment]
updated, and also removed manual org update in the stripeEvent hook for subscription.canceled

this should get taken care of by fetchCustomerAndSetSubscriptionDataOnOrg

File: api/src/coordinators/stripeEvent.js [hunk 2]
```diff
@@ -2,8 +2,12 @@ import { sendSingleRecipientEmail } from '../integrators/email'
 import stripeIntegrator from '../integrators/stripe'
 import BusinessOrganization from '../businesstime/organization'
 import billingCoordinator from '../coordinators/billing'
+<<<<<<< HEAD
```
Comment (dheniges) [inline api/src/coordinators/stripeEvent.js]: :jjidt:

File: api/src/coordinators/stripeEvent.js [hunk 3]
```diff
@@ -2,8 +2,12 @@ import { sendSingleRecipientEmail } from '../integrators/email'
...
+import { ORG_SUBSCRIPTION_STATUS } from '../constants/plans'
+=======
 import logger from '../integrators/logger'
 import { LOG_LEVELS } from '../constants/logging'
+>>>>>>> release
```
Comment (dheniges) [inline api/src/coordinators/stripeEvent.js]: :idt:

File: api/src/coordinators/stripeEvent.js [hunk 4]
```diff
@@ -2,8 +2,12 @@ import { sendSingleRecipientEmail } from '../integrators/email'
 import stripeIntegrator from '../integrators/stripe'
 import BusinessOrganization from '../businesstime/organization'
 import billingCoordinator from '../coordinators/billing'
+<<<<<<< HEAD
+import { ORG_SUBSCRIPTION_STATUS } from '../constants/plans'
+=======
```
Comment (dheniges) [inline api/src/coordinators/stripeEvent.js]: :dirksad:

**Review by dheniges — APPROVED** [review]
👍

**Pull request merged**